python/pyspark/sql/pandas/group_ops.py

```diff
@@ -501,7 +502,47 @@ def transformWithStateUDF(
                 )
 
             statefulProcessorApiClient.set_implicit_key(key)
-            result = statefulProcessor.handleInputRows(key, inputRows)
+
+            batch_timestamp = statefulProcessorApiClient.get_batch_timestamp()
+            watermark_timestamp = statefulProcessorApiClient.get_watermark_timestamp()
```

Comment:
We will need to have some values to initialize the `TimerValues` in the `handleInputRows`. On Scala side, we will always pass the real timestamp into `TimerValues` even timer is not defined: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala#L250

It is unlikely that users will call TimerValues if they do not have timer registered, but my original intention to save the call is to align the behavior with the Scala side. I guess here we might need to decide between saving a call and aligning with Scala side. I don't have a strong opinion on which is better. Which approach do you prefer?

Reply:
IIUC these 2 values are only being used when time mode is not `none`, I was meaning that for `none` time mode, we don't need these 2 extra API calls since it's not needed anyway